microraiden/microraiden/channel_manager/manager.py

```diff
@@ -348,6 +359,7 @@ def verify_balance_proof(self, sender, open_block_number, balance, signature):
 
     def register_payment(self, sender, open_block_number, balance, signature):
         """Register a payment."""
+        assert is_checksum_address(sender)
```

Comment:
I prefer converting address to checksummed one in public api. They should be liberal.

Reply:
I'm not entirely sure about how to handle this either. I think ethereum's overdue for making checksummed addresses mandatory, so I tried to enforce them in our internal, transaction-making classes here.

Simply converting all addresses to checksummed addresses defeats the purpose of the whole notion.

Reply:
Sure, internals it should be checksummed addr (as is the case in `blockchain.py` for example)
On the other hand, channel manager _may_ be used by some courageous fool to handle payments with a protocol other than HTTP, thus I'd say we can consider it a public API.
Having `is_checksum_address` now isn't a blocker of course, but we should be prepared to support other bizzare use cases in the future.